src/mc-fle2-payload-iev-v2.c

```diff
@@ -322,70 +300,99 @@ void mc_FLE2IndexedEncryptedValueV2_destroy(mc_FLE2IndexedEncryptedValueV2_t *ie
     _mongocrypt_buffer_cleanup(&iev->DecryptedServerEncryptedValue);
     _mongocrypt_buffer_cleanup(&iev->ServerEncryptedValue);
     _mongocrypt_buffer_cleanup(&iev->S_KeyId);
+
+    for (int i = 0; i < iev->edge_count; i++) {
+        mc_FLE2TagAndEncryptedMetadataBlock_cleanup(&iev->metadata[i]);
+    }
+
+    // Metadata array is dynamically allocated
+    bson_free(iev->metadata);
+
     bson_free(iev);
 }
 
-// -----------------------------------------------------------------------
-// Equality
+uint8_t mc_FLE2IndexedEncryptedValueV2_get_edge_count(const mc_FLE2IndexedEncryptedValueV2_t *iev,
+                                                      mongocrypt_status_t *status) {
+    assert(iev);
 
-bool mc_FLE2IndexedEqualityEncryptedValueV2_parse(mc_FLE2IndexedEncryptedValueV2_t *iev,
-                                                  const _mongocrypt_buffer_t *buf,
-                                                  mongocrypt_status_t *status) {
-    BSON_ASSERT_PARAM(iev);
-    BSON_ASSERT_PARAM(buf);
+    if (iev->type == kTypeInit) {
+        CLIENT_ERR("mc_FLE2IndexedEncryptedValueV2_get_edge_count "
+                   "must be called after "
+                   "mc_FLE2IndexedEncryptedValueV2_parse");
+        return 0;
+    }
 
-    if (iev->type != kTypeInit) {
-        CLIENT_ERR("mc_FLE2IndexedEqualityEncryptedValueV2_parse must not be "
-                   "called twice");
-        return false;
+    if (iev->type != kTypeRange) {
+        CLIENT_ERR("mc_FLE2IndexedEncryptedValueV2_get_edge_count must be called with type range");
+        return 0;
     }
 
-    mc_reader_t reader;
-    mc_reader_init_from_buffer(&reader, buf, __FUNCTION__);
+    return iev->edge_count;
+}
 
-    CHECK_AND_RETURN(mc_reader_read_u8(&reader, &iev->fle_blob_subtype, status));
+bool mc_FLE2IndexedEncryptedValueV2_get_edge(const mc_FLE2IndexedEncryptedValueV2_t *iev,
+                                             mc_FLE2TagAndEncryptedMetadataBlock_t *out,
+                                             const uint8_t edge_index,
+                                             mongocrypt_status_t *status) {
+    assert(iev);
```

Comment:
```suggestion
    BSON_ASSERT_PARAM (iev);
```

Suggest using `BSON_ASSERT_PARAM` instead of `assert`. `BSON_ASSERT_PARAM` produces a descriptive message on error:

```
The parameter: iev, in function mc_FLE2IndexedEncryptedValueV2_get_edge_count, cannot be NULL
```

Suggestion applies to other instances of `assert`.